examples/o1/Using_chained_calls_for_o1_structured_outputs.ipynb

```diff
@@ -0,0 +1,266 @@
+    "            \"reason\": \"JPMorgan Chase could benefit from AI by implementing advanced fraud detection systems, automating customer service with chatbots, enhancing risk management, and providing personalized financial advisory services using predictive analytics.\"\n",
+    "        },\n",
+    "        {\n",
+    "            \"company_name\": \"UnitedHealth Group\",\n",
+    "            \"page_link\": \"https://en.wikipedia.org/wiki/UnitedHealth_Group\",\n",
+    "            \"reason\": \"UnitedHealth Group could utilize AI to improve patient care through predictive analytics, automate administrative tasks, enhance diagnostic accuracy, and develop personalized treatment plans, thus increasing efficiency and patient satisfaction.\"\n",
+    "        }\n",
+    "    ]\n",
+    "}\n",
+    "```\n",
+    "\n",
+    "Note that the response is already quite good - it returns the JSON with the appropriate responses. However, it runs into the same pitfalls as existing use-cases of prompt-only JSON inference: \n",
+    "- You must manually process this JSON into your type-safe structure\n",
+    "- Model refusals are not [explicitly returned from the API as a separate structure](https://platform.openai.com/docs/guides/structured-outputs/refusals)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Structured Outputs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's now do this with [structured outputs.](https://platform.openai.com/docs/guides/structured-outputs). To enable this functionality, we'll chain our o1 response, with a 4o response that can effectively handle the data returned from the original o1 response. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "companies=[CompanyData(company_name='Walmart', page_link='https://en.wikipedia.org/wiki/Walmart', reason='As the largest retailer, Walmart can leverage AI for supply chain optimization, inventory management, personalized customer experiences, and operational efficiency.'), CompanyData(company_name='ExxonMobil', page_link='https://en.wikipedia.org/wiki/ExxonMobil', reason='ExxonMobil can benefit from AI in predictive maintenance, optimizing exploration and drilling, analyzing geological data, and enhancing safety measures.'), CompanyData(company_name='UnitedHealth Group', page_link='https://en.wikipedia.org/wiki/UnitedHealth_Group', reason='UnitedHealth Group can utilize AI to improve patient data analysis, diagnostics, personalize treatment plans, detect fraud, and increase operational efficiency.')]\n"
+     ]
+    }
+   ],
+   "source": [
+    "import requests\n",
+    "from pydantic import BaseModel\n",
+    "from openai import OpenAI\n",
+    "\n",
+    "client = OpenAI()\n",
```

Comment:
Do we need all of the same code that we had in the previous code cell? Can we reuse that and just apply the 4o chaining on top of it? Might make it a bit cleaner

Reply:
🙏

Reply:
Oh wait, I did this because I don't care about JSON schema adherence in the initial o1 call, so it can simplify the initial prompting.